app/models/user.rb

```diff
@@ -34,7 +34,7 @@ def pending_friends
 
   # Users who have requested to be friends
   def friend_requests
-    inverse_friendships.map { |friendship| friendship.user unless friendship.confirmed }.compact
+    inverse_friendships.map { |friendship| friendship unless friendship.confirmed }.compact
   end
 
   def confirm_friend(user)
```

Comment:
- [ ] When friendship is confirmed, an additional row with reversed user-friend pair is created
- [ ] Thanks to this method, we can find user friends only by user_id in friendships table
- `confirm_friend` can be defined as association with SQL query instead of processing objects in application memory (as the database was created to store and query data it is always more efficient than ruby)

```
has_many :confirmed_friendships, -> { where confirmed: true }, class_name: "Friendship"
has_many :friends, through: :confirmed_friendship

```